lib/docker.ts

```diff
@@ -101,7 +106,27 @@ class Docker {
     ]);
   }
 
-  public async pull(targetImage: string) {
+  public async pull(
```

Comment:
Would it make sense to put this split of `pull` and `pullCli` in separate commits? Feels like it can be its own self-contained change